public/pages/workflow_detail/workflow_detail.test.tsx

```diff
@@ -82,6 +92,86 @@ describe('WorkflowDetail', () => {
       expect(getByRole('tab', { name: 'Run queries' })).toBeInTheDocument();
       expect(getByRole('tab', { name: 'Errors' })).toBeInTheDocument();
       expect(getByRole('tab', { name: 'Resources' })).toBeInTheDocument();
+
+      // "Run ingestion" button should be enabled by default
+      const runIngestionButton = getByRole('button', { name: 'Run ingestion' });
+      expect(runIngestionButton).toBeInTheDocument();
+      expect(runIngestionButton).toBeEnabled();
+
+      // "Search pipeline >" button should be disabled by default
+      const searchPipelineButton = getByRole('button', {
+        name: 'Search pipeline >',
+      });
+      expect(searchPipelineButton).toBeInTheDocument();
+      expect(searchPipelineButton).toBeDisabled();
+
+      // "Create an ingest pipeline" option should be selected by default
+      const createIngestRadio = getByRole('radio', {
+        name: /Create an ingest pipeline Configure and ingest data into an index./,
+      });
+      expect(createIngestRadio).toBeChecked();
+
+      // "Skip ingestion pipeline" option should be disabled by default
+      const skipIngestRadio = getByRole('radio', {
+        name: /Skip ingestion pipeline Use an existing index with data ingested./,
+      });
+      expect(skipIngestRadio).not.toBeChecked();
+
+      // Clicking the "Export" button should open the export component
+      await waitFor(() =>
+        userEvent.click(getByRole('button', { name: 'Export' }))
+      );
+      expect(getByText('Export ' + workflowName)).toBeInTheDocument();
+      // Closing the "Export" opened above
+      await waitFor(() =>
+        userEvent.click(getByRole('button', { name: 'Close' }))
+      );
+
+      // Testing components in the ReactFlow workspace
+      const visualButton = getByRole('button', { name: 'Visual' });
+      expect(visualButton).toBeVisible();
+      expect(visualButton).toHaveClass('euiFilterButton-hasActiveFilters');
+      const jsonButton = getByRole('button', { name: 'JSON' });
+      expect(jsonButton).toBeVisible();
+      await waitFor(() => userEvent.click(jsonButton));
+      expect(jsonButton).toHaveClass('euiFilterButton-hasActiveFilters');
+
+      // Testing collapsible Tools panel
+      // Get the initial state of the Tools panel
+      const toolsPanelInitially = container.querySelector('#tools_panel_id');
+      expect(toolsPanelInitially).toBeVisible();
+      // Find and click the button to adjust panel sizes
+      const adjustPanelSizeButton = container.querySelector(
+        'button[aria-label="Press up or down to adjust panels size"].euiResizableButton--vertical'
+      );
```

Comment:
Overall this is awesome to see, thanks for all the additions! One suggestion for these types of query selection, is they can be more prone to break over time - for example, class names, internal text, underlying components, etc. may change due to changes in this codebase, or from the upstream UI libraries. One thing we can do is use the common pattern of applying a `data-test-subj` id to components we will test later on, that we have a static reference to a component that we have full control over.

For example, see the data-test-subj set [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/DetectorsList/components/EmptyMessage/EmptyMessage.tsx#L42)
And referenced in the test [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/DetectorsList/components/EmptyMessage/__tests__/EmptyMessage.test.tsx#L72)

I'd recommend seeing if some of the components pulled in here can be replaced with these types of references. Understandably, not all may be able to, as we have to check for class changes like `-isCollapsed` like you have on line 161.